### PR TITLE
refactor(console): extract manage language components

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

Extract manage language components:

- Rename the old `LanguageEditor` to `LanguageDetails`
- Add a new `LanguageEditor` to hold the language context
- Rename the old `ManageLanguageModal` to `LanguageEditorModal`
- Add a `ManageLanguageButton` to manage the `LanguageEditor` open state and used in the `LanguagesForm`.

## Testing

Test locally.

### CODE REVIEW

File: packages/console/src/pages/SignInExperience/components/ManageLanguage/LanguageEditor.tsx [hunk 1]
```diff
@@ -1,27 +1,29 @@
...
 import ConfirmModal from '@/components/ConfirmModal';
 import ModalLayout from '@/components/ModalLayout';
+import useUiLanguages from '@/hooks/use-ui-languages';
 import * as modalStyles from '@/scss/modal.module.scss';
 
-import { LanguageEditorContext } from '../../hooks/use-language-editor-context';
-import LanguageEditor from './LanguageEditor';
+import useLanguageEditorContext, {
+  LanguageEditorContext,
+} from '../../hooks/use-language-editor-context';
```
Comment (wangsijie) [inline packages/console/src/pages/SignInExperience/components/ManageLanguage/LanguageEditor.tsx:12]: the hook can be moved to the component folder?

File: packages/console/src/pages/SignInExperience/components/ManageLanguage/ManageLanguageButton.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+import LanguageEditor from './LanguageEditor';
+import * as styles from './ManageLanguageButton.module.scss';
+
+const ManageLanguageButton = () => {
+  const [isLanguageEditorOpen, setIsLanguageEditorOpen] = useState(false);
+
+  return (
+    <>
+      <Button
```
Comment (wangsijie) [inline packages/console/src/pages/SignInExperience/components/ManageLanguage/ManageLanguageButton.tsx:13]: can we remove this button from the "editor" component?

File: packages/console/src/pages/SignInExperience/components/ManageLanguage/LanguageEditor.tsx (hunk 1 above)
Comment (xiaoyijun) [reply packages/console/src/pages/SignInExperience/components/ManageLanguage/LanguageEditor.tsx]: updated.

**Review by wangsijie — APPROVED** [review]

File: packages/console/src/pages/SignInExperience/components/ManageLanguage/ManageLanguageButton.tsx (hunk 2 above)
Comment (xiaoyijun) [reply packages/console/src/pages/SignInExperience/components/ManageLanguage/ManageLanguageButton.tsx:13]: Reorganized.

**Pull request merged**